weather.py

```diff
@@ -1,98 +1,49 @@
+import json
 import requests
 from bs4 import BeautifulSoup
 
-# Function to fetch page content with error handling
-def get_page_content(url):
-    try:
-        page = requests.get(url)
-        page.raise_for_status()  # Raise HTTPError for bad responses
-        return page.content
-    except requests.exceptions.RequestException as e:
-        print(f"Error fetching page content: {e}")
+class WeatherScraper:
+    def __init__(self, cities_file):
+        self.cities = self.load_cities(cities_file)
+
+    def load_cities(self, file_path):
+        with open(file_path, 'r') as file:
+            return json.load(file)
+
+    def _get_page_content(self, url):
+        try:
+            page = requests.get(url)
+            page.raise_for_status()
+            return page.content
+        except requests.exceptions.RequestException as e:
+            print(f"Error fetching page content: {e}")
+            return None
+            
+    def _extract_weather_from_page(self, page_content):
+        if page_content:
+            soup = BeautifulSoup(page_content, "html.parser")
+            div = soup.find('div', id="seven-day-forecast-container")
+            if div:
+                li = div.find('li', class_="forecast-tombstone")
+                if li:
+                    p = li.find('p', class_="short-desc")
+                    if p:
+                        return f"\t{city}: {p.get_text()}"
```

Comment:
city variable is not defined, this method will error out.

Return only `p.get_text()` and don't print city name